### PR TITLE
feat(cli): add warning for incorrectly ordered flags

### LINKED ISSUE #16630: cli args - Show warning when putting permission flag after module name
Suggestion from @lucacasonato. It might be useful to show a warning when someone puts a permission flag after the module name. For example:

```
deno run main.js --allow-all
```

...that is incorrect and should be `deno run --allow-all main.js` because in the above it will instead provide `--allow-all` as an argument for `Deno.args` in main.js.

An issue with doing this might be that some apps expect having an argument like this though.

### PR DESCRIPTION
**Pull request by iuioiua**
This code checks if permission flags are incorrectly defined after the module name (e.g. `deno run mod.ts --allow-read` instead of the correct `deno run --allow-read mod.ts`). If so, a simple warning is displayed.

Is it worth adding a test for this case? If so, where's the best place for it?

Closes #16630. CC @bartlomieju

### CODE REVIEW

File: cli/main.rs
Comment (dsherret) [inline cli/main.rs:706]: Maybe `arg == "--allow-env"` because that won't require searching the entire `arg` string for each of these.

File: cli/main.rs
Comment (dsherret) [inline cli/main.rs:704]: We could avoid the allocations here by instead checking `flags.allow_all` etc.

File: cli/main.rs
Comment (dsherret) [inline cli/main.rs:713]: We could move this into a method on `Flags` itself because that will allow us to keep the knowledge of what the text of the arguments are (ex. `--allow-read`, etc.) buried within the flags.rs module (ex. `flags.has_permission()` and `flags.has_permission_in_argv()` or something like that).. It will also help keep the code of close beside `to_permission_args` and then we could write a quick unit test for this method.

**Review by dsherret — CHANGES_REQUESTED** [review]
Thanks! I left a few comments. We should probably move some of this code into `Flags` and write a unit test for that (ex. two separate methods like `flags.has_permission()` and `flags.has_permission_in_argv()` or something like that).

File: cli/main.rs
Comment (dsherret) [reply cli/main.rs:706]: Oh, true. Then `starts_with`

File: cli/main.rs
Comment (iuioiua) [reply cli/main.rs:713]: I usually prefer a functional approach. Is there any particular advantage over having a method instead of a simple function, or will a simple function suffice?

Comment (dsherret) [reply cli/main.rs:713]: One way is `has_permission(&flags)` the other is `flags.has_permission()`. The second way is much more discoverable.

File: cli/main.rs
Comment (bartlomieju) [inline cli/main.rs:707]: Looks good! Please add two integration tests to `run_tests.rs` - one that uses `--quiet` flag and one the doesn't (the former shouldn't print this warning), where you put permission flags after the script name. After that I think it will be good to land 👍

File: cli/main.rs
Comment (dsherret) [inline cli/main.rs:705]: We should probably refer to this as the "script argument" and not "module name" since that's what the help text calls it.

```
USAGE:
    deno run [OPTIONS] <SCRIPT_ARG>...
```

**Review by dsherret — APPROVED** [review]
LGTM once "module name" is renamed to match the help text as "script argument"

File: cli/main.rs
Comment (bartlomieju) [reply cli/main.rs:705]: If running a command like `deno run main.js --allow-read` the error message could read something like:
```
Permission flags have likely been incorrectly set after the module name. 

To grant permissions, set them before the module name:
  deno run --allow-read main.js
```

Comment (dsherret) [reply cli/main.rs]: I feel like we shouldn't add more logic that could go wrong.

Comment (bartlomieju) [reply cli/main.rs]: I think it's worth it for the DX of the user - seems like it could be easily tested with integration test?

Comment (dsherret) [reply cli/main.rs]: I don't think it's worth the extra complexity. A user should hopefully know what "script argument" means.

Comment (iuioiua) [reply cli/main.rs]: I agree. It's simple enough to change and makes the fix clearer.

Comment (dsherret) [reply cli/main.rs]: I really don't think we need more logic here. People will hopefully rarely hit this.

Comment (iuioiua) [reply cli/main.rs]: @dsherret, are you understanding that Bartek is suggesting that the changes should dynamically change according to the incorrect flag that the user has entered? If so, I get where you're coming from, but I don't think that's what he's saying... Is that right, @bartlomieju?

Comment (bartlomieju) [reply cli/main.rs]: I meant dynamic error, but a static one seems fine too - I think it's much clearer for people not yet familiar with Deno's permission system.

Comment (iuioiua) [reply cli/main.rs]: To clarify, I think this static text is the solution:
```
Permission flags have likely been incorrectly set after the script argument. 

To grant permissions, set them before the script argument:
    deno run [FLAG] main.js
```

Comment (dsherret) [reply cli/main.rs]: A static message sounds much better.

```
To grant permissions, set them before the script argument. For example:
    deno run --allow-read main.js
```

Comment (iuioiua) [reply cli/main.rs]: > A static message sounds much better.
> 
> ```
> To grant permissions, set them before the script argument. For example:
>     deno run --allow-read main.js
> ```

Yeah, I like that.

File: cli/tests/testdata/run/permission_args.out
Comment (dsherret) [inline cli/tests/testdata/run/permission_args.out:1]: We should remove the blank line here in the output.

Comment (dsherret) [reply cli/tests/testdata/run/permission_args.out:1]: @iuioiua I meant this blank line here at the start, not the other blank line. That one was good.

Comment (dsherret) [reply cli/tests/testdata/run/permission_args.out:1]: Oh, also we should remove the other blank line at the end of the warning. We don't normally surround warning with blank lines.

File: cli/main.rs
Comment (dsherret) [inline cli/main.rs:710]:
```suggestion
    log::warn!(
      r#"Permission flags have likely been incorrectly set after the script argument.

To grant permissions, set them before the script argument. For example:
    deno run --allow-read=. main.js"#
```

File: cli/tests/testdata/run/permission_args.out
Comment (dsherret) [reply cli/tests/testdata/run/permission_args.out:1]: Yup, we don't normally surround warnings with blank lines and the warning is in yellow so it should be differentiated by colour.

Comment (dsherret) [reply cli/tests/testdata/run/permission_args.out]: Sorry, I just realized this is not yellow so I pushed a change that makes it yellow to save you the hassle.

**Comment by dsherret** [comment]
Thanks a lot @iuioiua! LGTM. I'll merge this once the CI passes.

**Pull request merged**
